VirtualHavruta/vh.py

```diff
@@ -890,3 +1096,323 @@ def get_topic_descriptions(topic_slugs):
+            )
+            total_token_count += token_count
+            if len(candidate_chunks) > self.config["database"]["kg"]["max_depth"]:
+                candidate_chunks = candidate_chunks[:self.config["database"]["kg"]["max_depth"]]
+                candidate_rankings = candidate_rankings[:self.config["database"]["kg"]["max_depth"]]
+            elif len(candidate_chunks) == 0:
+                break
+        retrieval_res_kg = sorted(zip(collected_chunks, ranking_scores_collected_chunks), key=lambda pair: pair[1], reverse=True)
+
+        return retrieval_res_kg,  total_token_count
+
+    def get_linker_seed_chunks(self, linker_results: list[dict],
+                        filter_mode: str="primary", msg_id: str="") -> list[Document]:
+        """Given linker results, get the corresponding seed chunks.
+
+        First retrieve the seed nodes: linker results.
+        Find the chunks corresponding to the seed nodes. (1-to-many between nodes and chunks)
+
+        Parameters
+        ----------
+        linker_results
+            results from linker api
+        scripture_query
+            scripture query
+        msg_id
+            message id for slack, optional
+
+        Returns
+        -------
+            list of seed chunks
+        """
+        self.logger.info(f"MsgID={msg_id}. [RETRIEVAL] Starting get_linker_seed_chunks for KG search.")
+        seeds: list[Document] = self.retrieve_nodes_matching_linker_results(linker_results, msg_id, filter_mode=filter_mode)
+        seed_chunks: list[Document] = self.get_chunks_corresponding_to_nodes(seeds)
+        return seed_chunks
+
+    def rank_documents(self, chunks: list[Document], enriched_query: str, scripture_query: str|None=None, semantic_similarity_scores: list[float]|None = None,
+                              msg_id: str = "") -> list[Document]:
+        """Rank the document candidates in descending order based on their relevance to the query.
+
+        Return a new list, do not modify the input list.
+
+        Parameters
+        ----------
+        chunks
+            langchain documents
+        enriched_query
+            query enriched with additional context
+        scripture_query
+            query used to retrieve documents from the vector database
+        semantic_similarity_scores
+            optional, provide if available to save some computational costs
+        Returns
+        -------
+            ranked chunks
+        """
+        self.logger.info(f"MsgID={msg_id}. [RETRIEVAL] Starting rank_chunk_candidates for KG search.")
+        total_token_count = 0
+        if not semantic_similarity_scores:
+            if not scripture_query:
+                raise ValueError("Either provide semantic similarity scores or scripture query.")
+            semantic_similarity_scores: np.array = self.compute_semantic_similarity_documents_query(chunks, scripture_query)
+        reference_classes, token_count = self.get_reference_class(chunks, enriched_query)
```

Comment:
Primary references get reference class based on scripture_query.
Secondary references get reference class based on enriched_query.

Reply:
Status quo (prior to my MRs):
Both primary and secondary sources use enriched query. Should I adapt as you commented above?
It would require to update the parameters of the sort_reference method also in the private repo.

Reply:
https://github.com/Sefaria/AppliedAI-Private/blob/dev/app/slack_app.py#L167